add support for FreeBSD

fraenki opened the pull request:
This adds support for FreeBSD and handles the following differences:

* no 'fpm' package available (included in main PHP package)
* no 'devel' package available (also included in main PHP package)
* group 'root' is named 'wheel' on FreeBSD
* additional software is installed under '/usr/local'
* replace FreeBSD's system-wide `extensions.ini` with per-module configuration files

There are a few limitations though:

* no support for `manage_repos` on FreeBSD (not useful on this plattform)
* support for 'composer' and 'phpunit' is untested

Tested with the following hiera configuration:

```yaml
php::ensure: 'installed'
php::manage_repos: false
php::fpm: true
php::dev: true
php::composer: false
php::pear: true
php::phpunit: false
php::fpm::config:log_level: 'notice'
php::composer::auto_update: true
php::settings:
  Date/date.timezone: 'Europe/Berlin'
php::cli::settings:
  PHP/memory_limit: '256M'
php::fpm::settings:
  PHP/short_open_tag: 'On'
php::fpm::service::ensure: 'stopped'
php::fpm::service::enable: false
php::package_prefix: 'php55-'
php::extensions:
  bcmath: {}
  xml: {}
  mysql: {}
```

fpletz commented:
Thanks! Looks good to me. :beers: 

One question, though: You have set `php::fpm::service::enable` to `false`, but does managing the fpm service work? It might be a good idea to warn users if they want to use features that are not available on their platform.

fraenki commented:
@fpletz Thanks for your comments! I've updated my original submission:

* additions to run php-fpm service on FreeBSD
* replace FreeBSD's system-wide `extensions.ini` with per-module configuration files
* update `README.md` to mention known FreeBSD-specific limitations

Besides that I've tested php-fpm on FreeBSD and hence made some additions to `php-fpm.conf.erb`. I consider php-fpm on FreeBSD to be fully working now.

globin commented:
Cheers @fraenki!

Looks good to me, @fpletz feel free to merge.

mmoll commented:
I'd like to test it in my FreeBSD environment these days and merge then, if that's OK...

globin commented:
@mmoll ack

mmoll commented on README.md:
```diff
@@ -4,8 +4,8 @@
 on php-fpm. We strive to support all recent versions of Debian, Ubuntu,
-RedHat/CentOS and openSuSE/SLES. Managing Apache with `mod_php` is not
-supported.
+RedHat/CentOS, openSuSE/SLES and FreeBSD. Managing Apache with `mod_php`
```
it's "openSUSE" :snake:

fraenki replied:
fixed

mmoll commented on manifests/params.pp:
```diff
@@ -87,6 +89,29 @@
+      $fpm_service_name        = 'php-fpm'
+      $fpm_user                = 'www'
+      $fpm_group               = 'www'
+      $package_prefix          = 'php5-'
```
Not 100% sure, but ```php56-``` might be a better choice. I had problems on a clean system using pkgng binary packages with the defaults (php5 is 5.4, while the pear package requires 5.6)

mmoll commented:
I think we're having a slight ordering issue in this module that went maybe undetected on Linux:
```
root@bsdphp:~ # puppet apply --modulepath=/root/puppet/modules /root/puppet/lala.pp
Warning: Config file /usr/local/etc/puppet/hiera.yaml not found, using Hiera defaults
Notice: Compiled catalog for bsdphp.local in environment production in 5.76 seconds
Notice: /Stage[main]/Php/File[/usr/local/etc/php/extensions.ini]/ensure: removed
Notice: /Stage[main]/Php::Packages/Package[php56-extensions]/ensure: created
Notice: /Stage[main]/Php::Pear/Package[pear]/ensure: created
Notice: Finished catalog run in 7.76 seconds
root@bsdphp:~ # puppet apply --modulepath=/root/puppet/modules /root/puppet/lala.pp
Warning: Config file /usr/local/etc/puppet/hiera.yaml not found, using Hiera defaults
Notice: Compiled catalog for bsdphp.local in environment production in 5.67 seconds
Notice: /Stage[main]/Php/File[/usr/local/etc/php/extensions.ini]/ensure: removed
Notice: Finished catalog run in 1.94 seconds
root@bsdphp:~ # puppet apply --modulepath=/root/puppet/modules /root/puppet/lala.pp
Warning: Config file /usr/local/etc/puppet/hiera.yaml not found, using Hiera defaults
Notice: Compiled catalog for bsdphp.local in environment production in 5.67 seconds
Notice: Finished catalog run in 1.90 seconds
```

mmoll commented:
Please also add FreeBSD as supported OS in metadata.json

fraenki commented:
I've updated my original submission to include @mmoll's suggestions:

* bump default php version to 5.6 on FreeBSD (to conform with FreeBSD's new default version)
* fix ordering issue regarding 'extensions.ini' removal
* add FreeBSD as supported OS in 'metadata.json'
* rebased

mmoll commented:
Seems to work just fine here, thanks @fraenki! :ice_cream:

globin commented:
Awesome :tada:
